paywall/src/components/lock/Lock.js

```diff
@@ -22,7 +23,15 @@ export const Lock = ({
   config,
   disabled,
   hideModal,
+  openInNewWindow,
 }) => {
+  const { postMessage } = usePostMessage()
+  const purchase = key => {
```

Comment:
Wouldn't it make more sense to put that logic inside `mapStateToProps`? This would at least make it testable much more easily?

Reply:
Hooks cannot be outside of components. I could make a new custom hook if you'd like?